src/services/codefixes/fixExtendsInterfaceBecomesImplements.ts

```diff
@@ -27,11 +27,22 @@ namespace ts.codefix {
     }
 
     function doChanges(changes: textChanges.ChangeTracker, sourceFile: SourceFile, extendsToken: Node, heritageClauses: ReadonlyArray<HeritageClause>): void {
-        changes.replaceNode(sourceFile, extendsToken, createToken(SyntaxKind.ImplementsKeyword));
+        changes.replaceRange(sourceFile, { pos: extendsToken.getStart(), end: extendsToken.end }, createToken(SyntaxKind.ImplementsKeyword));
         // We replace existing keywords with commas.
         for (let i = 1; i < heritageClauses.length; i++) {

```

Comment:
It was this way before, but shouldn't this only apply to `heritageClauses[1]` since the others will have commas already?

Reply:
I thought that `implements I1, I2` counted as a single heritage clause (and that, in correct code, heritageClauses would have 0, 1, or 2 elements).

Reply:
OK, but then we don't need the loop, just an `if`.